### DISCUSSION TITLE
Assess bridge improvements

### DISCUSSION THREAD

**Discussion by mariocao** [witnet/witnet-solidity-bridge#198] [Ideas]
Previous results about gas cost showed that the complete data request flow could be decreased to make the bridge more viable given the current gas prices.

This assessment should cover some topics that should improve the bridge in terms of gas cost and usability:
- [ ] Optimistic functions (e.g. based on stakes and dispute mechanisms)
- [ ] Callbacks upon result reports
- [ ] Decrease contracts state (e.g. by providing state data on function calls)

Ideally, this assessment should include a high-level system design with some interface definitions and/or sequence diagrams.

**Comment by aesedepece** [comment]
I'll turn this now into an RFC discussion for now.
